src/components/color-picker/color-field/color-field.tsx

```diff
@@ -0,0 +1,437 @@
+      this.#drawColorField();
+    }
+  };
+
+  #drawColorField = (): void => {
+    const ctx = this.#canvasRef.getContext("2d", { willReadFrequently: true });
+    if (!ctx || !this.colorVariants) {
+      return;
+    }
+
+    const { width, height } = this.#canvasRef;
+
+    // Get the hue from the current color (HSV model)
+    const hsvValues = this.colorVariants.hsv;
+
+    // Clear the canvas
+    ctx.clearRect(0, 0, width, height);
+
+    // Create horizontal gradient from white to pure hue (saturation gradient)
+    const saturationGradient = ctx.createLinearGradient(0, 0, width, 0);
+    saturationGradient.addColorStop(0, `hsl(${hsvValues.h}, 0%, 100%)`);
+    saturationGradient.addColorStop(1, `hsl(${hsvValues.h}, 100%, 50%)`);
+
+    ctx.fillStyle = saturationGradient;
+    ctx.fillRect(0, 0, width, height);
+
+    // Apply vertical gradient from transparent to black (value gradient)
+    const valueGradient = ctx.createLinearGradient(0, 0, 0, height);
+    valueGradient.addColorStop(0, "rgba(0, 0, 0, 0)");
+    valueGradient.addColorStop(1, "rgba(0, 0, 0, 1)");
+
+    ctx.globalCompositeOperation = "multiply";
+    ctx.fillStyle = valueGradient;
+    ctx.fillRect(0, 0, width, height);
+
+    // Reset composite operation
+    ctx.globalCompositeOperation = "source-over";
+
+    this.#drawMarker();
+  };
+
+  #drawMarker = (): void => {
+    if (this.#markerRef) {
+      this.#markerRef.style.left = `${this.#currentX}px`;
+      this.#markerRef.style.top = `${this.#currentY}px`;
+    }
+  };
+
+  #updateMarkerPosition = (): void => {
+    if (!this.colorVariants || !this.#canvasRef) {
+      return;
+    }
+
+    // Use HSV for marker positioning
+    const hsvValues = this.colorVariants.hsv;
+    // X represents saturation (0-100%)
+    this.#currentX = (hsvValues.s / 100) * this.#canvasRef.width;
+    // Y represents value (0% value = bottom, 100% value = top)
+    this.#currentY = ((100 - hsvValues.v) / 100) * this.#canvasRef.height;
+    this.#drawMarker();
+  };
+
+  #updateCoordinates = (event: MouseEvent): void => {
+    const rect = this.#canvasRef.getBoundingClientRect();
+    const canvasHeight = this.#canvasRef.height;
+    const canvasWidth = this.#canvasRef.width;
+
+    // Ensure valid pixel range
+    this.#currentX = Math.max(
+      0,
+      Math.min(canvasWidth, event.clientX - rect.left)
+    );
+    this.#currentY = Math.max(
+      0,
+      Math.min(canvasHeight, event.clientY - rect.top)
+    );
+
+    this.#colorPickedVariants = this.#getColorFromCanvas();
+    this.#drawMarker();
+    // Use debounced update for drag operations
+    this.#updateStatusMessage();
+  };
+
+  #handleCanvasClick = (event: MouseEvent) => {
+    this.#updateCoordinates(event);
+    // Immediate update for single clicks
+    this.#updateStatusMessage(true);
+    this.input.emit(this.#colorPickedVariants);
+  };
+
+  #handleMouseDown = (event: MouseEvent) => {
+    this.#isDragging = true;
+    this.#updateCoordinates(event);
+  };
```

Comment:
Aren't we emitting twice the input event when we simply click on the color-field?